### PR TITLE
Move common fields to controller base class

### PR DESCRIPTION
**Pull request by dexamundsen**
- Move common fields to controller base class
- Move Azure specific functions from controller validation to azureUtils
- Move exceptions to appropriate package
- Minor error message updates
- No functional changes

Reason for change - some of the members are otherwise going to be redefined in AWS controllers.

### CODE REVIEW

File: service/src/main/java/bio/terra/workspace/app/controller/ControlledAzureResourceApiController.java [hunk 1]
```diff
@@ -485,7 +483,7 @@ public ResponseEntity<ApiCloneControlledAzureStorageContainerResult> cloneAzureS
         userRequest, workspaceId, body.getDestinationWorkspaceId(), resourceId);
     if (CloningInstructions.isReferenceClone(body.getCloningInstructions())) {
       throw new ValidationException(
-          "Copying azure storage containers by reference is not supported");
+          "Copying azure storage containers by `reference is not supported");
```
Comment (yuhuyoyo) [inline service/src/main/java/bio/terra/workspace/app/controller/ControlledAzureResourceApiController.java:486]: rm `

File: service/src/main/java/bio/terra/workspace/app/controller/ControlledResourceControllerBase.java [hunk 2]
```diff
@@ -43,16 +48,25 @@ public class ControlledResourceControllerBase extends ControllerBase {
...
-      SamService samService) {
-    super(authenticatedUserRequestFactory, request, samService);
+      ControlledResourceMetadataManager controlledResourceMetadataManager) {
+    super(authenticatedUserRequestFactory, request, samService, features, jobService, jobApiUtils);
     this.controlledResourceService = controlledResourceService;
+    this.controlledResourceMetadataManager = controlledResourceMetadataManager;
   }
 
-  public ControlledResourceService getControlledResourceService() {
+  public ControlledResourceService getControlledResourceService1() {
```
Comment (zloery) [inline service/src/main/java/bio/terra/workspace/app/controller/ControlledResourceControllerBase.java:62]: typo

File: service/src/main/java/bio/terra/workspace/common/utils/AzureUtils.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+
+/** Various utilities for validating requests in Azure VM Controllers. */
+public final class AzureUtils {
```
Comment (yuhuyoyo) [inline service/src/main/java/bio/terra/workspace/common/utils/AzureUtils.java:16]: AzureVMUtils?

File: service/src/main/java/bio/terra/workspace/app/controller/ControlledResourceControllerBase.java [hunk 4]
```diff
@@ -43,16 +48,25 @@ public class ControlledResourceControllerBase extends ControllerBase {
...
     return controlledResourceService;
   }
 
+  public ControlledResourceMetadataManager getControlledResourceMetadataManager1() {
```
Comment (zloery) [inline service/src/main/java/bio/terra/workspace/app/controller/ControlledResourceControllerBase.java:66]: here as well

**Review by yuhuyoyo — APPROVED** [review]

**Review by zloery — APPROVED** [review]

File: service/src/main/java/bio/terra/workspace/common/utils/AzureUtils.java (hunk 3 above)
Comment (dexamundsen) [reply service/src/main/java/bio/terra/workspace/common/utils/AzureUtils.java:16]: fixed the comment. It has non-VM utils as well

File: service/src/main/java/bio/terra/workspace/app/controller/ControlledAzureResourceApiController.java (hunk 1 above)
Comment (dexamundsen) [reply service/src/main/java/bio/terra/workspace/app/controller/ControlledAzureResourceApiController.java:486]: fixed

File: service/src/main/java/bio/terra/workspace/app/controller/ControlledResourceControllerBase.java (hunk 4 above)
Comment (dexamundsen) [reply service/src/main/java/bio/terra/workspace/app/controller/ControlledResourceControllerBase.java:66]: unused, removed

**Pull request merged**